physical/raft/raft.go

```diff
@@ -999,12 +999,18 @@ type RaftLock struct {
 func (l *RaftLock) monitorLeadership(stopCh <-chan struct{}, leaderNotifyCh <-chan bool) <-chan struct{} {
 	leaderLost := make(chan struct{})
 	go func() {
-		select {
-		case isLeader := <-leaderNotifyCh:
-			if !isLeader {
-				close(leaderLost)
+		for {
+			select {
+			case isLeader := <-leaderNotifyCh:
+				// if isLeader is true we want to continue to loop until we see
+				// a notification that we lost leadership.
```

Comment:
This comment can have tighter bounds here: `leaderNotifyCh` may deliver a `true` value initially if this server is already the leader prior to `RaftLock.Lock` call (the true message was already queued).  The next message is always going to be false.  So, the for-loop should loop twice at most.